extensions/notification/NotificationQueuePopulator.js

```diff
@@ -200,24 +312,29 @@ class NotificationQueuePopulator extends QueuePopulatorExtension {
      * @return {undefined} Promise|undefined
      */
     filterAsync(entry, cb) {
-        const { bucket, key, overheadFields } = entry;
+        if (this.notificationConfig.ignoreEmptyEvents && !entry.value) {
+            return cb();
```

Comment:
why do we we have a config `ignoreEmptyEvents` ?
- either this can happen & there is nothing to do → ignore them
- or it can happen & but there is nothing to do → ignore them
- or it can happen & there is something to do → we must not ignore them...

Reply:
Empty events have to be ignored in Zenko, else we'll get duplicate delete notifications. We use the update we do before each delete for delete notifications.
In S3C empty delete events are the only event available for delete notifications.

Reply:
is this filtering (in zenko) not already done in the mongo abstraction layers (e.g. either MongoClientInterface or LogReader), where we also do generate a "delete" event from an update with "deleted=true" ?

Reply:
Reading further, I see you removed this from `LogReader` : it seems to me this should not be "moved up" to QueuePopulator/..., but instead "moved down" into the specific implementations of `LogReader` (i.e. maybe a virtual function in LogReader, defined differently in each implementation ; or a flag from LogReader initialized by each subclass...)